src/gtfs/read.rs

```diff
@@ -199,12 +308,168 @@ pub fn read_stops<P: AsRef<path::Path>>(
     (stopareas, stoppoints)
 }
 
+#[derive(Eq, PartialEq)]
+enum RouteReadType {
+    RouteAsNtmRoute,
+    RouteAsNtmLine,
+}
+
+fn define_route_file_read_mode(gtfs_routes: &Vec<Route>) -> RouteReadType {
+    let mut iter = gtfs_routes.iter();
+    while let Some(r1) = iter.next() {
+        for r2 in iter.clone() {
+            if r1.is_same_line(r2) {
+                return RouteReadType::RouteAsNtmRoute;
+            }
+        }
+    }
+    return RouteReadType::RouteAsNtmLine;
+}
+
+fn get_commercial_mode_label(route_type: &RouteType) -> String {
+    use self::RouteType::*;
+    let result = match *route_type {
+        Tramway_LightRail => "Tram, Streetcar, Light rail",
+        Metro => "Subway, Metro",
+        Rail => "Rail",
+        Bus => "Bus",
+        Ferry => "Ferry",
+        CableCar => "Cable car",
+        Gondola_SuspendedCableCar => "Gondola, Suspended cable car",
+        Funicular => "Funicular",
+        Other(_) => "Unknown Mode",
+    };
+    result.to_string()
+}
+
+fn get_commercial_mode(route_type: &RouteType) -> objects::CommercialMode {
+    objects::CommercialMode {
+        id: route_type.to_gtfs_value(),
+        name: get_commercial_mode_label(&route_type),
+    }
+}
+
+fn get_physical_mode(route_type: &RouteType) -> objects::PhysicalMode {
+    use self::RouteType::*;
+    match *route_type {
+        Tramway_LightRail => objects::PhysicalMode {
+            id: "RailShuttle".to_string(),
+            name: "Navette ferrée (VAL)".to_string(),
+            co2_emission: None,
+        },
+        Metro => objects::PhysicalMode {
+            id: "Metro".to_string(),
+            name: "Métro".to_string(),
+            co2_emission: None,
+        },
+        Rail => objects::PhysicalMode {
+            id: "Train".to_string(),
+            name: "Train".to_string(),
+            co2_emission: None,
+        },
+        Bus => objects::PhysicalMode {
+            id: "Bus".to_string(),
+            name: "Bus".to_string(),
+            co2_emission: None,
+        },
+        Ferry => objects::PhysicalMode {
+            id: "Ferry".to_string(),
+            name: "Ferry".to_string(),
+            co2_emission: None,
+        },
+        CableCar | Gondola_SuspendedCableCar | Funicular => objects::PhysicalMode {
+            id: "Funicular".to_string(),
+            name: "Funicular".to_string(),
```

Comment:
Sometime in French, sometime in English

Reply:
you're right, it should be consistent. I'll make it all in English, sounds better for this piece of soft ;)